src/test/java/nl/knaw/dans/layerstore/LayeredItemStoreDeleteDirectoryTest.java

```diff
@@ -98,4 +103,29 @@ public void should_delete_directory_with_plain_file() throws Exception {
         );
         assertThat(list2).containsExactlyInAnyOrder("", "a");
     }
+
+    @Test
+    public void should_throw_cannot_not_delete_file_form_closed_layer() throws Exception {
+        var layerManager = new LayerManagerImpl(stagingDir, new ZipArchiveProvider(archiveDir), new DirectExecutor());
+        var layeredStore = new LayeredItemStore(dao, layerManager);
+        layeredStore.createDirectory("a/b/c/d");
+        layeredStore.writeFile("a/b/c/test.txt", toInputStream("Hello world!", UTF_8));
+        Files.createDirectories(archiveDir);
+        layerManager.newTopLayer();
+
+        // precondition: show database content
+        var list1 = daoTestExtension.inTransaction(() ->
+            dao.getAllRecords().toList().stream().map(ItemRecord::getPath)
+        );
+        assertThat(list1).containsExactlyInAnyOrder("", "a", "a/b", "a/b/c", "a/b/c/d", "a/b/c/test.txt");
+
+        // method under test
+        assertThatThrownBy(() -> layeredStore.deleteFiles(List.of("a/b/c/test.txt")))
+            .isInstanceOf(NoSuchFileException.class); // this is a failure for the wrong reason
```

Comment:
Do you mean, wrong but excepted, because there was nothing better than the `NoSuchFileException`?

Reply:
It should throw the exception after assumeNotYetFixed. As the design is still incomplete at this point, we just happen to get another failure.